metals/src/main/scala/scala/meta/internal/metals/Subject.scala

```diff
@@ -0,0 +1,12 @@
+package scala.meta.internal.metals
+
+trait Observable[T] {
+  def subscribe(fn: T => Unit): Unit
+}
+
+final class Subject[T] extends Observable[T] {
+  private var subscribers: List[T => Unit] = Nil
```

Comment:
it isn't thread-safe.
I would prefer to use just a function callback as long a there is no need in having multiply mutable subscribers

Reply:
> I would prefer to use just a function callback

I wanted to avoid that because this requires moving `private var doctor: Doctor = _` much earlier than it currently is. At some point of the time, we will have to get rid of the early initializations, but let future us worry about this :D